plugins/account_history_plugin/account_history_plugin.cpp

```diff
@@ -385,6 +384,25 @@ bool account_history_plugin_impl::is_scope_relevant(const eosio::types::vector<a
    return false;
 }
 
+fc::variant account_history_plugin_impl::transaction_to_variant(const signed_transaction& txn) const
+{
+   auto resolver = [&database = chain_plug->chain().get_database()]( const account_name& name ) -> optional<abi_serializer> {
+      const auto* accnt = database.find<chain::account_object,chain::by_name>( name );
+      if (accnt != nullptr) {
+         abi_def abi;
+         if (abi_serializer::to_abi(accnt->abi, abi)) {
+            return abi_serializer(abi);
+         }
+      }
+
+      return optional<abi_serializer>();
+   };
+
+   fc::variant pretty_output;
+   abi_serializer::to_variant(txn, pretty_output, resolver);
+   return pretty_output;
+}
+
```

Comment:
@wanderingbort What do you think about moving this to abi_serializer?

Reply:
Decision was to handle this in a separate ticket that will make sure that all transaction types can be converted to and from variants.